etna/datasets/internal_datasets.py

```diff
@@ -0,0 +1,135 @@
+    ----------
+    dataset_path:
+        path to dataset
+    """
+    return os.path.isfile(dataset_path)
+
+
+def _download_dataset_zip(url: str, file_name: str, **kwargs) -> pd.DataFrame:
+    """
+    Download zipped csv file.
+
+    Parameters
+    ----------
+    url:
+        url of the dataset
+    file_name:
+        csv file name in zip archive
+
+    Returns
+    -------
+    result:
+        dataframe with data
+
+    Raises
+    ------
+    Exception:
+        any error during downloading, saving and reading dataset from url
+    """
+    try:
+        with tempfile.TemporaryDirectory() as td:
+            temp_path = os.path.join(td, "temp.zip")
+            urllib.request.urlretrieve(url, temp_path)
+            with zipfile.ZipFile(temp_path) as f:
+                f.extractall(td)
+                df = pd.read_csv(os.path.join(td, file_name), **kwargs)
+    except Exception as err:
+        raise Exception(f"Error during downloading and reading dataset. Reason: {repr(err)}")
+    return df
+
+
+def load_dataset(name: str, download_path: Path = _DOWNLOAD_PATH, rebuild_dataset: bool = False) -> TSDataset:
+    """
+    Load internal dataset.
+
+    Parameters
+    ----------
+    name:
+        Name of the dataset.
+    download_path:
+        The path for saving dataset locally.
+    rebuild_dataset:
+        Whether to rebuild dataset from the initial source.
+
+    Returns
+    -------
+    result:
+        internal dataset
+
+    Raises
+    ------
+    NotImplementedError:
+        if name not from available list of dataset names
+    """
+    if name not in datasets_dict:
+        raise NotImplementedError(f"Dataset {name} is not available. You can use one from: {sorted(datasets_dict)}.")
+
+    dataset_dir = download_path / name
+    dataset_path = dataset_dir / f"{name}.csv"
+
+    get_dataset_function, freq = datasets_dict[name].values()
+    if not _check_dataset_local(dataset_path) or rebuild_dataset:
+        ts = get_dataset_function(dataset_dir)
+    else:
+        data = pd.read_csv(dataset_path)
+        ts = TSDataset(TSDataset.to_dataset(data), freq=freq)
+    return ts
+
+
+def get_electricity_dataset(dataset_dir) -> TSDataset:
+    """
+    Download6 save and load electricity dataset.
+    The electricity dataset is a 15 minutes time series of electricity consumption (in kW)
+    of 370 customers.
+
+    Parameters
+    ----------
+    dataset_dir:
+        The path for saving dataset locally.
+
+    Returns
+    -------
+    result:
+        electricity dataset in TSDataset format
+
+    References
+    ----------
+    .. [1] https://archive.ics.uci.edu/ml/datasets/ElectricityLoadDiagrams20112014
+    """
+    url = "https://archive.ics.uci.edu/static/public/321/electricityloaddiagrams20112014.zip"
+    os.makedirs(dataset_dir, exist_ok=True)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        data = _download_dataset_zip(url=url, file_name="LD2011_2014.txt", sep=";")
```

Comment:
What kind of warnings could you meet there?

Reply:
When we load `LD2011_2014.txt` via `pd.read_csv`, it returns huge warning message like this: 
```
DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18, ... ,368,369,370) have mixed types. Specify dtype option on import or set low_memory=False.
```

Reply:
Can't we specify dtypes somehow to avoid this?